8576 track granted motr to test 1761328033

paulo-filho-ctr opened the pull request:
8576 https://github.com/ustaxcourt/ef-cms/issues/8576

On the Case Information screen/Trial card, data is displayed indicating whether a case has been granted permission to proceed remotely (a simply yes/no or on/off flag will do) along with a date stamp indicating the date permission was granted.

The table `dw_case` requires two new columns `remote_trial_granted` and `remote_trial_granted_date`.

Link to the previous PR to gain context and where the comments are https://github.com/ustaxcourt/ef-cms/pull/9269#event-20489887869

paulo-filho-ctr committed:
8576: track granted motr

paulo-filho-ctr committed:
8576: add edit remote status sequences and modal

paulo-filho-ctr committed:
8576: add edit remote status sequences and modal

paulo-filho-ctr committed:
8576: add edit remote status sequences and modal

AnthonyLoeraTaxCourt committed:
8576: Updated validate calender sequences

paulo-filho-ctr committed:
8576: add optional clear button to modal dialog

paulo-filho-ctr committed:
9013: add max date to edit remote status to prevent set future dates

paulo-filho-ctr committed:
9013: add validation to edit remote status modal

paulo-filho-ctr committed:
9013: add validation to edit remote status modal

paulo-filho-ctr committed:
9013: edit remote status sends payload

paulo-filho-ctr committed:
9013: add clear date to edit remote status modal

AnthonyLoeraTaxCourt committed:
8576: updated clear sequence to clear dates on MOTR modal

paulo-filho-ctr committed:
8576: edit status modal clears date and submits valid dates

paulo-filho-ctr committed:
8576: add validation to edit remote status modal

paulo-filho-ctr committed:
8576: add validation to edit remote status modal

paulo-filho-ctr committed:
Merge remote-tracking branch 'origin/8576-track-granted-motr' into 8576-track-granted-motr

paulo-filho-ctr committed:
8576: save motr to database once migrations are run

paulo-filho-ctr committed:
8576: edit remote status modal persistence wired

AnthonyLoeraTaxCourt committed:
8576: Edited permissions access to edit remote staus modal

paulo-filho-ctr committed:
8576: styles based on permissions

paulo-filho-ctr committed:
8576: styles based on permissions

paulo-filho-ctr committed:
Merge remote-tracking branch 'origin/8576-track-granted-motr' into 8576-track-granted-motr

# Conflicts:
#	web-client/src/views/CaseDetail/CaseInformation/TrialInformation.tsx

paulo-filho-ctr committed:
8576: small styles improvements

paulo-filho-ctr committed:
8576: remove comment

paulo-filho-ctr committed:
8576: remove comment and unused file

paulo-filho-ctr committed:
8576: add test files

AnthonyLoeraTaxCourt committed:
8576: added test to test new functionality of Track Granted MOTR

AnthonyLoeraTaxCourt committed:
8567: fixed failing tests, and removed emoty test as well as cleaned up typescript errors

AnthonyLoeraTaxCourt committed:
8576: Resolved typescript errors

AnthonyLoeraTaxCourt committed:
Merge branch 'staging' into 8576-track-granted-motr

paulo-filho-ctr committed:
8576: clean up comments from test files

AnthonyLoeraTaxCourt committed:
Merge branch 'staging' into 8576-track-granted-motr

AnthonyLoeraTaxCourt committed:
8576: Updated code to resolve comments left on pr to test

AnthonyLoeraTaxCourt committed:
Fix type error: use undefined instead of null for remoteTrialGrantedDate

AnthonyLoeraTaxCourt committed:
8576: resolved type error

AnthonyLoeraTaxCourt committed:
Merge branch 'staging' into 8576-track-granted-motr

paulo-filho-ctr committed:
8576: edit remote status action call a case method to set the remote granted date

jimlerza committed:
Merge branch 'staging' into 8576-track-granted-motr

paulo-filho-ctr committed:
8576: update cypress tests and add return values to set remote trail granted date function

paulo-filho-ctr committed:
Merge branch '8576-track-granted-motr' into 8576-track-granted-motr-to-test-1761328033

jimlerza commented:
Not a comment on the code, exactly, but just to clarify - it is not required to manually run the `migration:postgres` script as this is run by CircleCI on every build.

TomElliottFlexion commented on web-api/elasticsearch/efcms-case-mappings.ts:
```diff
@@ -110,6 +110,12 @@ export const efcmsCaseMappings: Record<string, Object> = {
+    },
+    'remoteTrialGrantedDate.S': {
+      type: 'date',
+    },
```
Did we determine whether this needs to be indexed?

codyseibert replied:
I'm not seeing any code which is trying to query where remoteTrialGranded or remoteTrialGrantedDate, so for now I do not think an index is necessary.

akuny reviewed (changes requested):
According to the story, the ability to edit remote status should be limited to users in the Docket Clerk or Case Services Supervisor role.

When logged in as a trial clerk, I was able to open the modal and try to edit submit it, which resulted in a permissions failure on the backend.

akuny commented on shared/src/business/entities/cases/Case.ts:
```diff
@@ -1921,6 +1937,22 @@ export class Case extends JoiValidationEntity {
+   * @param {string | null | undefined} remoteTrialGrantedDate the date when motion to proceed remotely was granted, or null/undefined to clear
+   * @returns {Case} this case entity
+   */
+  setRemoteTrialGrantedDate(remoteTrialGrantedDate?: string | null): Case {
```
Can we please unit test this functionality?

paulo-filho-ctr replied:
Yes for sure.

paulo-filho-ctr committed:
8576: add unit test coverage

paulo-filho-ctr committed:
9292: add unit test coverage

paulo-filho-ctr committed:
8576: add conditional to edit remote status button

paulo-filho-ctr committed:
8576: add conditional to edit remote status button

codyseibert reviewed (approved):
great job

AnthonyLoeraTaxCourt committed:
8576: updates code due to fialing github actions

AnthonyLoeraTaxCourt committed:
Merge branch '8576-track-granted-motr' into 8576-track-granted-motr-to-test-1761328033

- Fixes isMemberCase import and test data structure
- Adds EditRemoteStatusButton to blocked from trial section
- Resolves edit remote status visibility issues in Cypress tests

AnthonyLoeraTaxCourt committed:
Fix EditRemoteStatusButton layout in Trial - Not Scheduled section

- Remove nested grid-col-6 class that was preventing proper display
- Update spacing with margin classes for better layout
- Fixes Cypress test failures for edit-remote-status button visibility

AnthonyLoeraTaxCourt committed:
removed unsued variables from opensearch, and resolved failing github actions

AnthonyLoeraTaxCourt committed:
8576: updated code in failing test

AnthonyLoeraTaxCourt committed:
Merge branch 'test' into 8576-track-granted-motr-to-test-1761328033

AnthonyLoeraTaxCourt committed:
Merge branch 'test' into 8576-track-granted-motr-to-test-1761328033

AnthonyLoeraTaxCourt committed:
reolved missing } from merge conflicts being resolved

AnthonyLoeraTaxCourt committed:
reolved added ts errors due to merge of testing into brnach with modifications

TomElliottFlexion commented on shared/src/business/entities/cases/Case.ts:
```diff
@@ -651,6 +655,18 @@ export class Case extends JoiValidationEntity {
+        'Whether the case has been granted a motion to proceed remotely.',
+      )
+      .meta({ tags: ['Restricted'] }),
+    remoteTrialGrantedDate: JoiValidationConstants.ISO_DATE.optional()
+      .allow(null)
+      .description(
+        'A date stamp indicating the date permission was granted to proceed remotely.',
+      ),
```
Not a big deal, but we could prevent a few edge cases by making this optional when `remoteTrialGranted` is undefined, but required when it is defined.
